### ISSUE TITLE
Support Naming Directories Where Fetch Contents will be Stored

### ISSUE THREAD

**Issue by danielhelfand** [carvel-dev/kapp-controller#151]
**Describe the problem/challenge you have**

As documented in this [code comment](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/858ec7e7713c5185bf571cfd5af720a47c8f035c/pkg/app/app_fetch.go#L27), fetched contents are stored in subdirectories since vendir doesn't allow placing contents in the vendir root. 

The current approach is to use a number starting at 0 and increment for each fetch source included. This isn't documented anywhere for users, so it can be difficult to determine where fetched contents are being stored. 

**Describe the solution you'd like**

To supplement the default behavior of creating/naming subdirectories (i.e. incrementing from 0 to n-1 fetch sources), I would like to be able to specify what the directory name for each fetch source should be. 

```yaml
fetch:
  - git:
      url: https://github.com/k14s/kapp
      ref: origin/develop
      subPath: examples/gitops/guestbook
      subDirName: myDirNameGit
  - http:
      url: https://raw.githubusercontent.com/k14s/kapp/db2cc63e12e988235eb8815af8edc0ca0cfaa79c/examples/simple-app-example/config-1.yml
      sha256: 71b0a2dceb6bb6e7a519d0587abd9400a265cabbe2c084f783df94a92db6d980
      subDirName: myDirNameHttp
template:
  - ytt:
        paths:
          - myDirNameGit/foo.yml
          - myDirNameHttp/bar.yml
```

By default though, kapp-controller could keep its current approach of creating/naming sub directories for those not concerned with where sources are stored.

**Comment by benmoss** [comment]
@neil-hickey is there anything more to do here? didn't #544 fix this?

**Comment by neil-hickey** [comment]
Fixed by #544 - released in 0.33.1
